components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/nats/InboundNatsListener.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 The Apache Software Foundation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.inbound.endpoint.protocol.nats;
+import org.wso2.carbon.inbound.endpoint.protocol.nats.management.NatsEndpointManager;
+
+import org.apache.synapse.inbound.InboundProcessorParams;
+import org.apache.synapse.inbound.InboundRequestProcessor;
+
+/**
+ * Listener class for NATS Inbound Endpoint which is triggered by inbound core
+ */
+public class InboundNatsListener implements InboundRequestProcessor {
+
+    private String name;
+    private InboundProcessorParams processorParams;
+
+    public InboundNatsListener(InboundProcessorParams params) {
+        processorParams = params;
+        name = params.getName();
+    }
+
+    @Override public void init() {
+        NatsEndpointManager.getInstance().startEndpoint(0, name, processorParams);
+    }
+
+    @Override public void destroy() {
+        NatsEndpointManager.getInstance().closeEndpoint(0);
```

Comment:
what is we rename closeEndpoint to stopEndpoint. Start and stop seems to go along together :)

Reply:
closeEndpoint is an overridden method from the superclass